enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/ChoreoAnalyticsProvider.java

```diff
@@ -224,11 +224,8 @@ private String getValueAsString(Map<String, Value> fieldsMap, String key) {
     }
 
     private Map<String, Value> getFieldsMapFromLogEntry() {
-        if (logEntry.getCommonProperties() == null
-                || logEntry.getCommonProperties().getMetadata() == null
-                || logEntry.getCommonProperties().getMetadata().getFilterMetadataMap() == null
-                || !logEntry.getCommonProperties().getMetadata().getFilterMetadataMap()
-                .containsKey(MetadataConstants.EXT_AUTH_METADATA_CONTEXT_KEY)) {
+        if (!logEntry.getCommonProperties().getMetadata().getFilterMetadataMap()
+                        .containsKey(MetadataConstants.EXT_AUTH_METADATA_CONTEXT_KEY)) {
```

Comment:
In [this commit](https://github.com/wso2/product-microgateway/commit/b07322e36ce9e0078c610a022b1680b24041dac1) you have intentionally added those null checks. I am wondering maybe this was because the proto objects can still be null even if the IDE says that the null check id redundant.

Reply:
I deleted because the IDE suggests that it can't be null.
And when I check the code, I could see that the null check is included within the function.
To keep the changes to a minimal level, I will add them again then